fmriprep/workflows/bold/outputs.py

```diff
@@ -256,60 +256,133 @@ def init_func_derivatives_wf(
+            run_without_submitting=True, mem_gb=DEFAULT_MEMORY_MIN_GB)
 
         workflow.connect([
             (inputnode, select_fs_surf, [
                 ('surf_files', 'surfaces'),
                 ('surf_refs', 'keys')]),
-            (select_fs_surf, name_surfs, [('surfaces', 'in_file'),
-                                          ('surf_kwargs', 'template_kwargs')]),
+            (select_fs_surf, name_surfs, [('surfaces', 'in_file')]),
             (inputnode, ds_bold_surfs, [('source_file', 'source_file')]),
-            (select_fs_surf, ds_bold_surfs, [('surfaces', 'in_file')]),
-            (name_surfs, ds_bold_surfs, [('out_name', 'suffix')]),
+            (select_fs_surf, ds_bold_surfs, [('surfaces', 'in_file'),
+                                             ('key', 'space')]),
+            (name_surfs, ds_bold_surfs, [('hemi', 'hemi')]),
         ])
 
     # CIFTI output
     if cifti_output:
-        from niworkflows.interfaces.cifti import CiftiNameSource
-
-        name_cifti = pe.MapNode(
-            CiftiNameSource(), iterfield=['variant', 'density'], name='name_cifti',
-            mem_gb=DEFAULT_MEMORY_MIN_GB, run_without_submitting=True)
-        cifti_bolds = pe.MapNode(
-            DerivativesDataSink(base_directory=output_dir, compress=False),
-            iterfield=['in_file', 'suffix'], name='cifti_bolds',
-            run_without_submitting=True, mem_gb=DEFAULT_MEMORY_MIN_GB)
-        cifti_key = pe.MapNode(DerivativesDataSink(
-            base_directory=output_dir), iterfield=['in_file', 'suffix'],
-            name='cifti_key', run_without_submitting=True,
+        ds_bold_cifti = pe.Node(DerivativesDataSink(
+            base_directory=output_dir, suffix='bold', compress=False),
+            name='ds_bold_cifti', run_without_submitting=True,
             mem_gb=DEFAULT_MEMORY_MIN_GB)
         workflow.connect([
-            (inputnode, name_cifti, [('cifti_variant', 'variant'),
-                                     ('cifti_density', 'density')]),
-            (inputnode, cifti_bolds, [('bold_cifti', 'in_file'),
-                                      ('source_file', 'source_file')]),
-            (name_cifti, cifti_bolds, [('out_name', 'suffix')]),
-            (name_cifti, cifti_key, [('out_name', 'suffix')]),
-            (inputnode, cifti_key, [('source_file', 'source_file'),
-                                    ('cifti_metadata', 'in_file')]),
+            (inputnode, ds_bold_cifti, [(('bold_cifti', _unlist), 'in_file'),
+                                        ('source_file', 'source_file'),
+                                        (('cifti_metadata', _get_surface), 'space'),
+                                        ('cifti_density', 'density'),
+                                        (('cifti_metadata', _read_json), 'meta_dict')])
         ])
 
     return workflow
+
+
+def init_bold_preproc_report_wf(mem_gb, reportlets_dir, name='bold_preproc_report_wf'):
```

Comment:
This change is a bit out of scope. This workflow is not used anywhere (I wrote it when there were plans to show a before/after reportlet without/with preprocessing. This will potentially be used at some point, and I thought it was more findable here in the outputs module.